opsmngr/feature_control_policies_test.go

```diff
@@ -0,0 +1,260 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opsmngr
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/go-test/deep"
+)
+
+func TestFeatureControlPoliciesServiceOp_List(t *testing.T) {
+	client, mux, teardown := setup()
+	defer teardown()
+
+	path := fmt.Sprintf("/groups/%s/controlledFeature", groupID)
+
+	mux.HandleFunc(path, func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, http.MethodGet)
+		_, _ = fmt.Fprint(w, `{
+			  "_id": "5c8100bcf2a30b12ff88258f",
+			  "created": "2019-08-29T15:03:24Z",
+			  "updated": "2019-08-29T15:03:24Z",
+			  "externalManagementSystem":
+			  {
+				"name": "Operator",
+				"systemId": "6d6c139ae5528707b6e8e3b2",
+				"version": "0.2.1"
+			  },
+			  "policies": [
+				{"policy": "ExternallyManagedLock"},
+				{"policy": "DisableUserManagement"},
+				{"policy": "DisableAuthenticationMechanisms"},
+				{"policy": "DisableSetMongodVersion"},
+				{
+				  "policy": "DisableSetMongodConfig",
+				  "disabledParams": ["net.tls.CAFile"]
+				}
+			  ]
+			}`)
+	})
+
+	logs, _, err := client.FeatureControlPolicies.List(ctx, groupID, nil)
+	if err != nil {
+		t.Fatalf("FeatureControlPolicies.List returned error: %v", err)
+	}
+
+	expected := &FeaturePolicy{
+		ID:      groupID,
+		Created: "2019-08-29T15:03:24Z",
+		Updated: "2019-08-29T15:03:24Z",
+		ExternalManagementSystem: &ExternalManagementSystem{
+			Name:     "Operator",
+			SystemID: "6d6c139ae5528707b6e8e3b2",
+			Version:  "0.2.1",
+		},
```

Comment:
Reading the docs, I don't see the required values are captured. We only require the System name, the rest are optional.

Upto you, as the client already will return the error on our behalf, but you might want to test the scenario.

for reference, the validation is done here: https://github.com/10gen/mms/blob/master/server/src/main/com/xgen/module/controlledfeature/svc/ControlledFeatureSvc.java#L152-L163